ddp-workspace/projects/toolkit/src/lib/components/dialogs/sessionWillExpire.component.ts

```diff
@@ -68,13 +72,20 @@ export class SessionWillExpireComponent implements OnInit, OnDestroy {
 
     public signOut(): void {
         this.renewNotifier.hideSessionExpirationNotifications();
-        this.session.disableTokenExpiredProcess();
         this.auth0.logout();
     }
 
     public renewSession(): void {
-        this.renewNotifier.hideSessionExpirationNotifications();
-        this.session.disableTokenExpiredProcess();
+        // Blocks UI to prevent interference in the session renewing process
+        this.isRenewing = true;
+        const session = this.session.sessionObservable.pipe(
+            // Because the session is BehaviorSubject, we should skip a current session,
+            // we are only interested in the renewed session
+            skip(1)
+        ).subscribe(() => {
+            this.renewNotifier.hideSessionExpirationNotifications();
+        });
+        this.anchor.add(session);
         this.auth0.auth0RenewToken();
     }
```

Comment:
I am that flushing explicitly the previous session object is necessary! If this is indeed the case, can we push down this functionality into out of the dialog and down into one of the services?

Reply:
Idea - get the last session object after session renewing. Why we should move it to another service? I guess this action is specific only for this case. Or did I misunderstand you?

Reply:
Was asking for this to be moved down to a service because this looks like token renewal implementation/back-end plumbing on session management rather than a high-level API call that would be visible to a UI component.

Reply:
Sorry, but I'm not sure that we are talking about the same thing... Because we just watching for the session. Did you pull the last changes? (I'm sorry if you did, I'm a bit confused)
![image](https://user-images.githubusercontent.com/30126907/79467132-1fce6300-8006-11ea-8ce8-4899577d0dd8.png)

Reply:
I think we are talking about same thing. To be more explicit, my expectation would be that this component simply calls a `renewSession()` on one of our services associated with session management. e.g., `Auth0AdapterService` and whatever needs to happen to this correctly happens there instead of in the sessionwillexpire dialog.

Reply:
Okay, thanks for the explanation.